File: CodeTraining/LeetCode/easy/linklist/Code_237.java
package LeetCode.easy.linklist;

public class Code_237 {

    private class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }
}

File: CodeTraining/LeetCode/easy/linklist/code_237.scala
package LeetCode.easy.linklist

object code_237 {

  private class ListNode(var _x: Int = 0) {
    var next: ListNode = null
    var x: Int = _x
  }

  def deleteNode(node: ListNode): Unit = {
    node.x = node.next.x
    node.next = node.next.next
  }

}
